runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxPanelManager.java

```diff
@@ -0,0 +1,191 @@
+		Provider<ChatboxTextMenuInput> chatboxTextMenuInputProvider, Provider<ChatboxTextInput> chatboxTextInputProvider)
+	{
+		this.client = client;
+		this.clientThread = clientThread;
+		this.eventBus = eventBus;
+
+		this.keyManager = keyManager;
+		this.mouseManager = mouseManager;
+
+		this.chatboxTextMenuInputProvider = chatboxTextMenuInputProvider;
+		this.chatboxTextInputProvider = chatboxTextInputProvider;
+
+		eventBus.register(this);
+	}
+
+	public void close()
+	{
+		clientThread.invokeLater(this::unsafeCloseInput);
+	}
+
+	void unsafeCloseInput()
+	{
+		client.runScript(
+			ScriptID.RESET_CHATBOX_INPUT,
+			0,
+			1
+		);
+	}
+
+	void unsafeOpenInput(ChatboxInput input)
+	{
+		client.runScript(ScriptID.CLEAR_CHATBOX_PANEL);
+
+		eventBus.register(input);
+		if (input instanceof KeyListener)
+		{
+			keyManager.registerKeyListener((KeyListener) input);
+		}
+		if (input instanceof MouseListener)
+		{
+			mouseManager.registerMouseListener((MouseListener) input);
+		}
+		if (input instanceof MouseWheelListener)
+		{
+			mouseManager.registerMouseWheelListener((MouseWheelListener) input);
+		}
+
+		currentInput = input;
+		client.setVar(VarClientInt.INPUT_TYPE, -3);
+		client.getWidget(WidgetInfo.CHATBOX_TITLE).setHidden(true);
+		client.getWidget(WidgetInfo.CHATBOX_FULL_INPUT).setHidden(true);
+
+		Widget c = getContainerWidget();
+		clearDynamicChildren(c);
+		c.setOnDialogAbortListener((JavaScriptCallback) ev -> this.unsafeCloseInput());
+		input.open();
+	}
+
+	public void openInput(ChatboxInput input)
+	{
+		clientThread.invokeLater(() -> unsafeOpenInput(input));
+	}
+
+	public ChatboxTextMenuInput openTextMenuInput(String title)
+	{
+		return chatboxTextMenuInputProvider.get()
+			.title(title);
+	}
+
+	public ChatboxTextInput openTextInput(String prompt)
+	{
+		return chatboxTextInputProvider.get()
+			.prompt(prompt);
+	}
+
+	@Subscribe
+	public void onScriptCallbackEvent(ScriptCallbackEvent ev)
+	{
+		switch (ev.getEventName())
+		{
+			case "resetChatboxInput":
+				if (currentInput != null)
+				{
+					try
+					{
+						currentInput.close();
+					}
+					catch (Exception e)
+					{
+						log.warn("Exception closing {}", currentInput.getClass(), e);
+					}
+
+					eventBus.unregister(currentInput);
+					if (currentInput instanceof KeyListener)
+					{
+						keyManager.unregisterKeyListener((KeyListener) currentInput);
+					}
+					if (currentInput instanceof MouseListener)
+					{
+						mouseManager.unregisterMouseListener((MouseListener) currentInput);
+					}
+					if (currentInput instanceof MouseWheelListener)
+					{
+						mouseManager.unregisterMouseWheelListener((MouseWheelListener) currentInput);
+					}
+				}
+				currentInput = null;
+		}
+	}
+
+	public Widget getContainerWidget()
+	{
+		return client.getWidget(WidgetInfo.CHATBOX_CONTAINER);
+	}
+
+	public static void clearDynamicChildren(Widget c)
+	{
+		if (c.getChildren() != null)
+		{
+			Arrays.fill(c.getChildren(), null);
```

Comment:
Wouldn't c.setChildren(null) be better?

Reply:
It may work, this is what the cs2 opcode does to I would prefer to do that. Also this saves a few allocations.